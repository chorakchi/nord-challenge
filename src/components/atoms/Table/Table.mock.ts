export const mockData = {
  header: [
    {
      label: "Name",
      key: "name",
    },
    {
      label: "Distance",
      key: "distance",
    },
  ],

  items: [
    {
      name: "Germany #21",
      distance: 431,
    },
    {
      name: "Singapore #70",
      distance: 581,
    },
    {
      name: "Germany #76",
      distance: 566,
    },
    {
      name: "United States #67",
      distance: 1815,
    },
    {
      name: "Singapore #91",
      distance: 191,
    },
    {
      name: "United Kingdom #48",
      distance: 199,
    },
    {
      name: "United States #45",
      distance: 151,
    },
    {
      name: "Latvia #95",
      distance: 1562,
    },
    {
      name: "Germany #39",
      distance: 1604,
    },
    {
      name: "Japan #64",
      distance: 1584,
    },
    {
      name: "Singapore #68",
      distance: 567,
    },
    {
      name: "United Kingdom #41",
      distance: 502,
    },
    {
      name: "Lithuania #48",
      distance: 110,
    },
    {
      name: "Latvia #26",
      distance: 1862,
    },
    {
      name: "United Kingdom #68",
      distance: 1069,
    },
    {
      name: "Japan #62",
      distance: 1769,
    },
    {
      name: "United States #57",
      distance: 1503,
    },
    {
      name: "Germany #56",
      distance: 834,
    },
    {
      name: "Japan #70",
      distance: 752,
    },
    {
      name: "United States #39",
      distance: 414,
    },
    {
      name: "United Kingdom #38",
      distance: 994,
    },
    {
      name: "Latvia #86",
      distance: 668,
    },
    {
      name: "Latvia #35",
      distance: 260,
    },
    {
      name: "Singapore #36",
      distance: 19,
    },
    {
      name: "Singapore #56",
      distance: 947,
    },
    {
      name: "Germany #86",
      distance: 697,
    },
    {
      name: "Singapore #26",
      distance: 1423,
    },
    {
      name: "United States #98",
      distance: 654,
    },
    {
      name: "Latvia #42",
      distance: 178,
    },
    {
      name: "Singapore #40",
      distance: 1329,
    },
  ],
};
