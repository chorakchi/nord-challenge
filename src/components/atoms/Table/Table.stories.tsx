import { ComponentStory, ComponentMeta } from "@storybook/react";
import { mockData } from "./Table.mock";

import { Table } from "./Table";

export default {
  title: "Atoms/Table",
  component: Table,
  parameters: {
    docs: {
      description: {
        component:
          `Table component display information in a table view format of rows and columns..<br/>` +
          `<br/> <br/> <h5> Form props:</h5>` +
          " this component had props for heading and items. and it accepting array for them."
      },
    },
    backgrounds: {
      default: "light",
    },
  },
} as ComponentMeta<typeof Table>;

const Template: ComponentStory<typeof Table> = (args) => (
  <Table {...args} />
);

export const Table_ = Template.bind({});
Table_.args = {
  ...mockData
};
