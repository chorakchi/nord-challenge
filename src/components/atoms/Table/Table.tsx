import { TypographyProps } from "./Table.interface";
export const Table: React.FC<TypographyProps> = ({
  children,
  header,
  onClickHeader,
  items,
  ...props
}: TypographyProps) => {
  return (
    <table {...props} className="relative min-w-full divide-y divide-gray-200">
      <thead className="sticky">
        <tr>
          {header.map((item) => (
            <th
              onClick={() => {
                onClickHeader(item.key);
              }}
              key={'th'+item.key}
              scope="col"
              className="sticky bg-white top-0 px-6 py-3 text-left text-xs font-medium text-gray-500 uppercase tracking-wider hover:bg-purple-50 cursor-pointer"
            >
              {item.label}
            </th>
          ))}
        </tr>
      </thead>
      <tbody className="bg-white divide-y divide-gray-200">
        {items.map((country, i) => (
          <tr key={'tr'+i}>
            {header.map((item, j) => (
              <td key={'tdf'+j} className="px-6 py-4 whitespace-nowrap">
                <div className="text-sm text-gray-900">{country[item.key]}</div>
              </td>
            ))}
          </tr>
        ))}
      </tbody>
    </table>
  );
};
