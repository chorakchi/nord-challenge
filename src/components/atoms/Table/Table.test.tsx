import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import { Table } from "./Table";
import { mockData } from "./Table.mock";



describe("Typography", () => {
  it("should render the Typography", () => {
    const subject = shallow(<Table {...mockData} />);
    expect(subject).toHaveLength(1);
  });

  it("should render correctly and match to the Snapshot", () => {
    const subject = renderer.create(<Table {...mockData} />).toJSON();
    expect(subject).toMatchSnapshot();
  });
});
