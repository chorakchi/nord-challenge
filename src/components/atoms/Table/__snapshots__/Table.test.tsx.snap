// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography should render correctly and match to the Snapshot 1`] = `
<table
  className="relative min-w-full divide-y divide-gray-200"
>
  <thead
    className="sticky"
  >
    <tr>
      <th
        className="sticky bg-white top-0 px-6 py-3 text-left text-xs font-medium text-gray-500 uppercase tracking-wider hover:bg-purple-50 cursor-pointer"
        onClick={[Function]}
        scope="col"
      >
        Name
      </th>
      <th
        className="sticky bg-white top-0 px-6 py-3 text-left text-xs font-medium text-gray-500 uppercase tracking-wider hover:bg-purple-50 cursor-pointer"
        onClick={[Function]}
        scope="col"
      >
        Distance
      </th>
    </tr>
  </thead>
  <tbody
    className="bg-white divide-y divide-gray-200"
  >
    <tr>
      <td
        className="px-6 py-4 whitespace-nowrap"
      >
        <div
          className="text-sm text-gray-900"
        >
          Germany #21
        </div>
      </td>
      <td
        className="px-6 py-4 whitespace-nowrap"
      >
        <div
          className="text-sm text-gray-900"
        >
          431
        </div>
      </td>
    </tr>
    <tr>
      <td
        className="px-6 py-4 whitespace-nowrap"
      >
        <div
          className="text-sm text-gray-900"
        >
          Singapore #70
        </div>
      </td>
      <td
        className="px-6 py-4 whitespace-nowrap"
      >
        <div
          className="text-sm text-gray-900"
        >
          581
        </div>
      </td>
    </tr>
    <tr>
      <td
        className="px-6 py-4 whitespace-nowrap"
      >
        <div
          className="text-sm text-gray-900"
        >
          Germany #76
        </div>
      </td>
      <td
        className="px-6 py-4 whitespace-nowrap"
      >
        <div
          className="text-sm text-gray-900"
        >
          566
        </div>
      </td>
    </tr>
    <tr>
      <td
        className="px-6 py-4 whitespace-nowrap"
      >
        <div
          className="text-sm text-gray-900"
        >
          United States #67
        </div>
      </td>
      <td
        className="px-6 py-4 whitespace-nowrap"
      >
        <div
          className="text-sm text-gray-900"
        >
          1815
        </div>
      </td>
    </tr>
    <tr>
      <td
        className="px-6 py-4 whitespace-nowrap"
      >
        <div
          className="text-sm text-gray-900"
        >
          Singapore #91
        </div>
      </td>
      <td
        className="px-6 py-4 whitespace-nowrap"
      >
        <div
          className="text-sm text-gray-900"
        >
          191
        </div>
      </td>
    </tr>
    <tr>
      <td
        className="px-6 py-4 whitespace-nowrap"
      >
        <div
          className="text-sm text-gray-900"
        >
          United Kingdom #48
        </div>
      </td>
      <td
        className="px-6 py-4 whitespace-nowrap"
      >
        <div
          className="text-sm text-gray-900"
        >
          199
        </div>
      </td>
    </tr>
    <tr>
      <td
        className="px-6 py-4 whitespace-nowrap"
      >
        <div
          className="text-sm text-gray-900"
        >
          United States #45
        </div>
      </td>
      <td
        className="px-6 py-4 whitespace-nowrap"
      >
        <div
          className="text-sm text-gray-900"
        >
          151
        </div>
      </td>
    </tr>
    <tr>
      <td
        className="px-6 py-4 whitespace-nowrap"
      >
        <div
          className="text-sm text-gray-900"
        >
          Latvia #95
        </div>
      </td>
      <td
        className="px-6 py-4 whitespace-nowrap"
      >
        <div
          className="text-sm text-gray-900"
        >
          1562
        </div>
      </td>
    </tr>
    <tr>
      <td
        className="px-6 py-4 whitespace-nowrap"
      >
        <div
          className="text-sm text-gray-900"
        >
          Germany #39
        </div>
      </td>
      <td
        className="px-6 py-4 whitespace-nowrap"
      >
        <div
          className="text-sm text-gray-900"
        >
          1604
        </div>
      </td>
    </tr>
    <tr>
      <td
        className="px-6 py-4 whitespace-nowrap"
      >
        <div
          className="text-sm text-gray-900"
        >
          Japan #64
        </div>
      </td>
      <td
        className="px-6 py-4 whitespace-nowrap"
      >
        <div
          className="text-sm text-gray-900"
        >
          1584
        </div>
      </td>
    </tr>
    <tr>
      <td
        className="px-6 py-4 whitespace-nowrap"
      >
        <div
          className="text-sm text-gray-900"
        >
          Singapore #68
        </div>
      </td>
      <td
        className="px-6 py-4 whitespace-nowrap"
      >
        <div
          className="text-sm text-gray-900"
        >
          567
        </div>
      </td>
    </tr>
    <tr>
      <td
        className="px-6 py-4 whitespace-nowrap"
      >
        <div
          className="text-sm text-gray-900"
        >
          United Kingdom #41
        </div>
      </td>
      <td
        className="px-6 py-4 whitespace-nowrap"
      >
        <div
          className="text-sm text-gray-900"
        >
          502
        </div>
      </td>
    </tr>
    <tr>
      <td
        className="px-6 py-4 whitespace-nowrap"
      >
        <div
          className="text-sm text-gray-900"
        >
          Lithuania #48
        </div>
      </td>
      <td
        className="px-6 py-4 whitespace-nowrap"
      >
        <div
          className="text-sm text-gray-900"
        >
          110
        </div>
      </td>
    </tr>
    <tr>
      <td
        className="px-6 py-4 whitespace-nowrap"
      >
        <div
          className="text-sm text-gray-900"
        >
          Latvia #26
        </div>
      </td>
      <td
        className="px-6 py-4 whitespace-nowrap"
      >
        <div
          className="text-sm text-gray-900"
        >
          1862
        </div>
      </td>
    </tr>
    <tr>
      <td
        className="px-6 py-4 whitespace-nowrap"
      >
        <div
          className="text-sm text-gray-900"
        >
          United Kingdom #68
        </div>
      </td>
      <td
        className="px-6 py-4 whitespace-nowrap"
      >
        <div
          className="text-sm text-gray-900"
        >
          1069
        </div>
      </td>
    </tr>
    <tr>
      <td
        className="px-6 py-4 whitespace-nowrap"
      >
        <div
          className="text-sm text-gray-900"
        >
          Japan #62
        </div>
      </td>
      <td
        className="px-6 py-4 whitespace-nowrap"
      >
        <div
          className="text-sm text-gray-900"
        >
          1769
        </div>
      </td>
    </tr>
    <tr>
      <td
        className="px-6 py-4 whitespace-nowrap"
      >
        <div
          className="text-sm text-gray-900"
        >
          United States #57
        </div>
      </td>
      <td
        className="px-6 py-4 whitespace-nowrap"
      >
        <div
          className="text-sm text-gray-900"
        >
          1503
        </div>
      </td>
    </tr>
    <tr>
      <td
        className="px-6 py-4 whitespace-nowrap"
      >
        <div
          className="text-sm text-gray-900"
        >
          Germany #56
        </div>
      </td>
      <td
        className="px-6 py-4 whitespace-nowrap"
      >
        <div
          className="text-sm text-gray-900"
        >
          834
        </div>
      </td>
    </tr>
    <tr>
      <td
        className="px-6 py-4 whitespace-nowrap"
      >
        <div
          className="text-sm text-gray-900"
        >
          Japan #70
        </div>
      </td>
      <td
        className="px-6 py-4 whitespace-nowrap"
      >
        <div
          className="text-sm text-gray-900"
        >
          752
        </div>
      </td>
    </tr>
    <tr>
      <td
        className="px-6 py-4 whitespace-nowrap"
      >
        <div
          className="text-sm text-gray-900"
        >
          United States #39
        </div>
      </td>
      <td
        className="px-6 py-4 whitespace-nowrap"
      >
        <div
          className="text-sm text-gray-900"
        >
          414
        </div>
      </td>
    </tr>
    <tr>
      <td
        className="px-6 py-4 whitespace-nowrap"
      >
        <div
          className="text-sm text-gray-900"
        >
          United Kingdom #38
        </div>
      </td>
      <td
        className="px-6 py-4 whitespace-nowrap"
      >
        <div
          className="text-sm text-gray-900"
        >
          994
        </div>
      </td>
    </tr>
    <tr>
      <td
        className="px-6 py-4 whitespace-nowrap"
      >
        <div
          className="text-sm text-gray-900"
        >
          Latvia #86
        </div>
      </td>
      <td
        className="px-6 py-4 whitespace-nowrap"
      >
        <div
          className="text-sm text-gray-900"
        >
          668
        </div>
      </td>
    </tr>
    <tr>
      <td
        className="px-6 py-4 whitespace-nowrap"
      >
        <div
          className="text-sm text-gray-900"
        >
          Latvia #35
        </div>
      </td>
      <td
        className="px-6 py-4 whitespace-nowrap"
      >
        <div
          className="text-sm text-gray-900"
        >
          260
        </div>
      </td>
    </tr>
    <tr>
      <td
        className="px-6 py-4 whitespace-nowrap"
      >
        <div
          className="text-sm text-gray-900"
        >
          Singapore #36
        </div>
      </td>
      <td
        className="px-6 py-4 whitespace-nowrap"
      >
        <div
          className="text-sm text-gray-900"
        >
          19
        </div>
      </td>
    </tr>
    <tr>
      <td
        className="px-6 py-4 whitespace-nowrap"
      >
        <div
          className="text-sm text-gray-900"
        >
          Singapore #56
        </div>
      </td>
      <td
        className="px-6 py-4 whitespace-nowrap"
      >
        <div
          className="text-sm text-gray-900"
        >
          947
        </div>
      </td>
    </tr>
    <tr>
      <td
        className="px-6 py-4 whitespace-nowrap"
      >
        <div
          className="text-sm text-gray-900"
        >
          Germany #86
        </div>
      </td>
      <td
        className="px-6 py-4 whitespace-nowrap"
      >
        <div
          className="text-sm text-gray-900"
        >
          697
        </div>
      </td>
    </tr>
    <tr>
      <td
        className="px-6 py-4 whitespace-nowrap"
      >
        <div
          className="text-sm text-gray-900"
        >
          Singapore #26
        </div>
      </td>
      <td
        className="px-6 py-4 whitespace-nowrap"
      >
        <div
          className="text-sm text-gray-900"
        >
          1423
        </div>
      </td>
    </tr>
    <tr>
      <td
        className="px-6 py-4 whitespace-nowrap"
      >
        <div
          className="text-sm text-gray-900"
        >
          United States #98
        </div>
      </td>
      <td
        className="px-6 py-4 whitespace-nowrap"
      >
        <div
          className="text-sm text-gray-900"
        >
          654
        </div>
      </td>
    </tr>
    <tr>
      <td
        className="px-6 py-4 whitespace-nowrap"
      >
        <div
          className="text-sm text-gray-900"
        >
          Latvia #42
        </div>
      </td>
      <td
        className="px-6 py-4 whitespace-nowrap"
      >
        <div
          className="text-sm text-gray-900"
        >
          178
        </div>
      </td>
    </tr>
    <tr>
      <td
        className="px-6 py-4 whitespace-nowrap"
      >
        <div
          className="text-sm text-gray-900"
        >
          Singapore #40
        </div>
      </td>
      <td
        className="px-6 py-4 whitespace-nowrap"
      >
        <div
          className="text-sm text-gray-900"
        >
          1329
        </div>
      </td>
    </tr>
  </tbody>
</table>
`;
